### PR TITLE
Feat: Add script translations to bundle

### PR DESCRIPTION
**Pull request by Dathix**
Proposal for #273

### CODE REVIEW

File: src/Roots/Acorn/Assets/Concerns/Enqueuable.php [hunk 1]
```diff
@@ -189,6 +189,26 @@ public function localize($name, $object)
...
+     */
+    public function translate($domain, $path)
+    {
+        if (! $handles = $this->js()->keys()->toArray()) {
+            return $this;
+        }
+
+        foreach ($handles as $handle) {
+            wp_set_script_translations("{$this->id}/{$handle}", $domain, $path);
+        }
```
Comment (QWp6t) [inline src/Roots/Acorn/Assets/Concerns/Enqueuable.php:207]:
```suggestion
        $this->js()->keys()->each(function ($handle) {
            wp_set_script_translations("{$this->id}/{$handle}", $domain, $path);
        }
```

**Review by QWp6t — COMMENTED** [review]
I don't have much experience with i18n. I see room for a minor improvement in the code unless you disagree. I'll wait for someone else on the team more familiar with WordPress i18n to actually test this.

**Comment by QWp6t** [comment]
Should `$path` default to `lang_path()`?
Should `$domain` default to `wp_get_theme()->get('TextDomain')`?

This is outside my wheelhouse so I'll let others determine whether defaults are a good idea. Just throwing that out there as an idea.

**Comment by Dathix** [comment]
I like the idea of having these defaults.

**Comment by Dathix** [comment]
@QWp6t I have added the defaults. Would love to get this feature merged as soon as possible.

File: src/Roots/Acorn/Assets/Concerns/Enqueuable.php [hunk 2]
```diff
@@ -189,6 +189,25 @@ public function localize($name, $object)
...
+     * Add script translations to be used by the bundle
+     *
+     * @param string $domain
+     * @param string $path
+     * @return $this
+     */
+    public function translate($domain = null, $path = null)
+    {
+        $domain = $domain ?? wp_get_theme()->get('TextDomain');
+        $path = $path ?? lang_path();
```
Comment (QWp6t) [inline src/Roots/Acorn/Assets/Concerns/Enqueuable.php:202]:
```suggestion
        $domain ??= wp_get_theme()->get('TextDomain');
        $path ??= lang_path();
```

**Comment by Dathix** [comment]
@QWp6t Thanks!

**Review by retlehs — APPROVED** [review]

**Pull request merged**